src/frequenz/sdk/microgrid/component/_component.py

```diff
@@ -8,9 +8,42 @@
 from dataclasses import dataclass
 from enum import Enum
 
+import frequenz.api.microgrid.inverter_pb2 as inverter_pb
 import frequenz.api.microgrid.microgrid_pb2 as microgrid_pb
 
 
+class ComponentType(Enum):
+    """A ABC from which individual component types are derived."""
+
+
+class InverterType(ComponentType):
```

Comment:
Any reason to derive from an empty `Enum`?

Reply:
Just for typing.  Then we can have a single `type` field in the `Component` object, and assign `InverterType` or `MeterType` values to it and mypy will still be happy.  That way we'd also get an interface that follows the protobuf definition somewhat.

Reply:
But what I don't understand is, if you are using a method that returns `ComponentType`, then you can't access any of the `InverterType` enum members, right? This is the part I don't understand

Reply:
it doesn't return a `ComponentType`.  it returns an `InverterType` or `MeterType`, etc.  The type hint says `ComponentType`  and that keeps mypy happy,  because the return values do implement `ComponentType` (because it is empty).